src/Chunk.cpp

```diff
@@ -0,0 +1,103 @@
+/**
+*  @file    Chunk.cpp
+*  @author  
+*  @date    05 June 2018
+*/
+
+#include "Chunk.h"
+#include "Perlin.h"
+
+#define Z_VALUE 0
+
+Chunk::Chunk(u8 x, u8 y, Area areas[], u8 type) : x(x * CHUNK_SIZE), y(y * CHUNK_SIZE), type(type)
+{
+	// Ich bin irgendwie zu dumm das richtig zu machen XD
+    this->areas[0] = areas[0];
+    this->areas[1] = areas[1];
+    this->areas[2] = areas[2];
+    this->areas[3] = areas[3];
+
+	if (type < 0 || type > 3)
+		throw std::invalid_argument("Illegal chunk type - must be in between 0 and 3");
+}
+
+void Chunk::calculate() // Hier k�nnte man evtl. Z �bergeben, falls man sp�ter doch wieder animieren will..
+{
+	switch (type)
+	{
+		case 0: calculate_inner(); break;
+		case 1: calculate_vertical(); break;
+		case 2: calculate_horizontal(); break;
+		case 3: calculate_corner(); break;
+		default: throw std::invalid_argument("Illegal chunk type - must be in between 0 and 3"); break;
```

Comment:
Je nach Typ wird dann die entsprechende Methode aufgerufen... wichtig, vor allem auch der Unterschied zwischen Horizontal und Vertikal, weil wir da ja die Werte entweder in Abhängigkeit von X oder Y interpolieren müssen.